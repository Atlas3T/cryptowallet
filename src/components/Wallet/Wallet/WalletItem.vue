<template>
  <div
    class="coin"
    @click="selectCoin(data.uid)"
  >
    <div class="details">
      <div class="icon"/>
      <div class="name">{{ data.name }}</div>
    </div>
    <div class="balance">
      <div>{{ data.balance }}</div>
      <div>
        <Amount
          :amount="data.balance"
          :prepend-plus-or-minus="false"
          :currency="currency"
          :coin="data.key"
        />
      </div>
    </div>
    <div>
      <q-icon name="chevron_right"/>
    </div>
  </div>
</template>

<script>
import Amount from '@/components/Wallet/SharedComponents/Amount.vue';

export default {
  name: 'WalletItem',
  components: {
    Amount,
  },
  props: {
    data: {
      type: Object,
      required: true,
    },
    currency: {
      type: String,
      required: true,
    },
  },
  methods: {
    selectCoin(uid) {
      this.$router.push({ path: `/wallet/balance/${uid}` });
    },
  },
};
</script>

<style scoped>
.coin {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid #13273a;
}

.coin .details {
  display: flex;
  align-items: center;
}

.icon {
  border-radius:100%;
  background: #41678a;
  width: 2.5rem;
  height: 2.5rem;
  margin-right: 1rem;
}
</style>
