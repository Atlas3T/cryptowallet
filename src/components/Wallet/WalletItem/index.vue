<template>
  <div
    class="wallet selectWallet"
    @click="selectWallet(wallet.uid)"
  >
    <div class="details">
      <div class="icon"/>
      <div class="name">{{ wallet.name }}</div>
    </div>
    <div class="balance">
      <div>{{ wallet.balance }}</div>
      <Amount
        :amount="wallet.balance"
        :prepend-plus-or-minus="false"
        :currency="currency"
        :coin="wallet.key"
      />
    </div>
    <div>
      <q-icon name="chevron_right"/>
    </div>
  </div>
</template>

<script>
import Amount from '@/components/Wallet/Amount';

export default {
  name: 'WalletItem',

  components: {
    Amount,
  },

  props: {
    wallet: {
      type: Object,
      required: true,
    },
    currency: {
      type: Object,
      required: true,
    },
  },

  methods: {
    selectWallet(uid) {
      this.$router.push({ path: `/wallet/balance/${uid}` });
    },
  },
};
</script>

<style scoped>
.coin {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  border-bottom: 1px solid #13273a;
}

.coin .details {
  display: flex;
  align-items: center;
}

.icon {
  border-radius:100%;
  background: #41678a;
  width: 2.5rem;
  height: 2.5rem;
  margin-right: 1rem;
}
</style>
