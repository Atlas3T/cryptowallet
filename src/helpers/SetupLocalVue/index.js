import Quasar, * as All from 'quasar';
import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandler from '@/plugins/ErrorHandler';

const localVue = createLocalVue();

localVue.use(VueI18n);
localVue.use(Vuex);
localVue.use(Quasar, { components: All, directives: All, plugins: All });
localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandler({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact-item/:id',
      },
    ],
  });

  if (store) sync(store, router);
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
