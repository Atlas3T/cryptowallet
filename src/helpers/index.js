import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
