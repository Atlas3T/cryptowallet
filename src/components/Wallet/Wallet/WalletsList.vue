<template>
  <section class="scroll-area">
    <q-scroll-area
      class="scroll-area"
    >
      <WalletItem
        v-for="coin in coins"
        :key="coin.name"
        :data="coin"
        :currency="currency"
      />
    </q-scroll-area>
  </section>
</template>

<script>
import WalletItem from './WalletItem.vue';

export default {
  name: 'WalletsList',
  components: {
    WalletItem,
  },
  props: {
    coins: {
      type: Object,
      required: true,
    },
    currency: {
      type: String,
      required: true,
    },
  },
};
</script>

<style scoped>
.scroll-area {
  position: absolute;
  height: 100%;
  width: 100%;
  left: 0;
  top: 0;
}
</style>
