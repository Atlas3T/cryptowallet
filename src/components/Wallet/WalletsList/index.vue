<template>
  <section class="scroll-area">
    <div v-if="wallets.length === 0">
      <q-btn
        icon="add_circle_outline"
        label="add"
        color="primary"
        size="xl"
        class="large-cloud-btn"
        @click.prevent="openWalletsModal"
      />
    </div>

    <q-scroll-area
      v-if="wallets.length > 0"
      class="scroll-area"
    >
      <WalletItem
        v-for="wallet in wallets"
        :key="wallet.displayName"
        :wallet="wallet"
        :currency="currency"
        :click-item-action="clickItemAction"
      />
    </q-scroll-area>
  </section>
</template>

<script>
import WalletItem from '@/components/Wallet/WalletItem';
import Coin from '@/store/wallet/entities/coin';

export default {
  name: 'WalletsList',
  components: {
    WalletItem,
  },
  props: {
    // wallets: {
    //   type: Array,
    //   required: true,
    // },

    currency: {
      type: Object,
      required: false,
    },

    clickItemAction: {
      type: String,
      required: true,
    },
  },
  computed: {
    wallets() {
      console.log('Coin.all() :', Coin.all());
      return Coin.all();
    },
  },
  methods: {
    openWalletsModal() {
      this.$root.$emit('walletsModalOpened', true);
    },
  },
};
</script>

<style>
.scroll-area {
  position: absolute;
  height: 100%;
  width: 100%;
  left: 0;
  top: 0;
}

.large-cloud-btn {
  width: 100%;
  height: 3em;
  border-radius: 0.3em;
  font-size: 3em;
  padding: 0em;
  color: #1e3c57;
}

.large-cloud-btn i {
  color: #1e3c57;
}

.large-cloud-btn .q-btn-inner{
  color: #1e3c57;
}
</style>
