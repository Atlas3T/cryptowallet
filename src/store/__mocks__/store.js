import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
