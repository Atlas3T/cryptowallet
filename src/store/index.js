/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
