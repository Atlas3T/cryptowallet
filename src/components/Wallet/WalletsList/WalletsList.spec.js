import { shallowMount } from '@vue/test-utils';
import WalletsList from '@/components/Wallet/WalletsList';
import WalletItem from '@/components/Wallet/WalletItem';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';

describe('WalletsList.vue', () => {
  let storeMocks;
  let wrapper;
  let store;

  const propsData = {
    wallets: {
      BTC: {
        key: 'BTC',
        name: 'Bitcoin',
        balance: 25.532344,
        uid: 1,
      },
      ATL: {
        key: 'ATL',
        name: 'Atlas',
        balance: 21.532344,
        uid: 2,
      },
      ETH: {
        key: 'ETH',
        name: 'Ethereum',
        balance: 55.532344,
        uid: 3,
      },
    },
    currency: {},
    wallets: {},
  };

  function wrapperInit (options) {
    return shallowMount(WalletsList, options);
  }

  function storeInit (custom) {
    storeMocks = createStoreMocks(custom);
    wrapper = wrapperInit({ propsData, localVue, i18n, store: storeMocks.store });
    store = wrapper.vm.$store;
  }

  beforeEach(() => storeInit());

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a section element with .scroll-area class', () => {
    expect(wrapper.contains('section.scroll-area')).toBe(true);
  });

  it('renders a correct number of WalletItem components', () => {
    expect(wrapper.findAll(WalletItem).length === Object.keys(propsData.wallets).length).toBe(true);
  });
});
