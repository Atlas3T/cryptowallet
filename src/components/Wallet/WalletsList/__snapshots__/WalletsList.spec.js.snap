// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletsList.vue renders and matches snapshot 1`] = `
<section
  class="scroll-area"
>
  <qscrollarea-stub
    class="scroll-area"
    contentactivestyle="[object Object]"
    contentstyle="[object Object]"
    delay="1000"
    thumbstyle="[object Object]"
  />
</section>
`;
